File: json_to_flutter/functions/insert/insert_in.dart
import '../../models/index.dart';
import '../checkRelationAndInsert.dart';
import '../costBenefit.dart';
import '../insert/insert.dart';

insertInside(No father, Widget widget) {
  if (father.children.length == 0) {
    father.children.add(new No(father, widget, []));
  } else {
    No inside;
    // Checking if it's inside something inside
    for (var child in father.children) {
      Relation vd = checkIfIsInside(child.widget, widget);
      if (vd == Relation.stack || vd == Relation.inside) {
        inside = child;
        break;
      }
    }
    if (inside != null) {
      // It's inside something inside
      checkRelationAndInsert(inside, widget);
    } else {
      // It isn't inside something inside
      // Cost benefit to know where put
      CostBenefit cost = costBenefit(father, widget, onlyInside: true);
      No putOn = cost.bestPosition == 0
          ? father
          : father.children[cost.bestPosition - 1];
      insert(putOn, widget, cost.type);
    }
  }
}

File: flutter_code_generate/src/functions/widget/insert/insert_in.js
const { checkIfIsInside, checkRelationAndInsert } = require("../checkRelationAndInsert");
const { insertWidgetAtNoWithType } = require("./insert");
const { No } = require("../models/no");

function insertInside(father, widget) {
    if (father.children.length == 0) {
        father.children.push(new No(father, widget, []));
    } else {
        let inside;
        // Checking if it's inside something inside
        for (let i = 0; i < father.children.length; i++) {
            let child = father.children[i];
            let vd = checkIfIsInside(child.widget, widget, false, false);
            if (vd == Relation.STACK || vd == Relation.INSIDE) {
                inside = child;
                break;
            }
        }
        if (inside != null) {
            // It's inside something inside
            checkRelationAndInsert(inside, widget);
        } else {
            // It isn't inside something inside
            // Cost benefit to know where put
            let cost = costBenefit(father, widget, true);
            let putOn = cost.bestPosition == 0
                ? father
                : father.children[cost.bestPosition - 1];
            insertWidgetAtNoWithType(widget, putOn, cost.type);
        }
    }
}

module.exports = { insertInside };

